### PR TITLE
[user] Add getSites functions in User class

### LINKED ISSUE #9064: [user_accounts] Outdated TestPlan.md + potential permission issue
The [TestPlan.md](https://github.com/aces/Loris/blob/87a3dd5bde053728b99db49cc194de8a96886370/modules/user_accounts/test/TestPlan.md?plain=1#L44) says the following:

> 15. Verify that if the editor does not have permission 'Across all sites add and edit users' then the site drop-down list is populated with the editor's associated sites, otherwise all sites are displayed.

`Site` is now a multi-select and this should be reflected in the documentation.

Note: The above quoted item is only true for the `Edit User` page. When at the index, the `FilterableDataTable`'s `Site` filter contains all sites even if the user only has permissions for a single site and the `User Accounts: View/Create/Edit User Accounts - Own Sites` permission. This does not appear the be the case for other modules' `FilterableDataTable`.

### PR DESCRIPTION
**Pull request by CamilleBeau**
## Brief summary of changes
This PR adds 2 functions to the user class. One returns an array of Site objects that the user has access to. The other returns an associative array with the format CenterID => SiteName for all of the sites that the user has access to.

- [x] Have you updated related documentation?

### CODE REVIEW

File: modules/user_accounts/php/user_accounts.class.inc [hunk 1]
```diff
@@ -77,9 +77,24 @@ class User_Accounts extends \DataFrameworkMenu
...
         ];
+        // Get user
+        $factory = \NDB_Factory::singleton();
+        $user    = $factory->user();
+
+        // Without user_acounts_multisite permission, only show user sites
+        if ($user->hasPermission('user_accounts_multisite')) {
+            $sites = \Utility::getSiteList(false);
+        } else {
+            $sites = [];
```
Comment (jeffersoncasimir) [inline modules/user_accounts/php/user_accounts.class.inc:88]: I believe you should be using `$user->getStudySites()` instead of reimplementing the logic

**Review by jeffersoncasimir — CHANGES_REQUESTED** [review]

File: modules/user_accounts/php/user_accounts.class.inc (hunk 1 above)
Comment (CamilleBeau) [reply modules/user_accounts/php/user_accounts.class.inc:88]: @jeffersoncasimir getStudySites only includes study sites, but I believe in this case we want _all_ sites that the user has access to

**Comment by CamilleBeau** [comment]
@jeffersoncasimir @driusan (and @ridz1208 because you seem interested in this development) I added 2 functions to the User class: 
1. getSites() that returns an array of Site objects for all sites that the user has
2. getSitesNamesList() that returns an associative array of centerID => centerName

I'm still trying to set up Docker on my VM to be able to debug the test suites (it was failing yesterday), but I'm trying something out in my last commit.

**Review by jeffersoncasimir — APPROVED** [review]
This has been tested and works as expected. Great work!

File: modules/user_accounts/php/useraccountrowprovisioner.class.inc [hunk 2]
```diff
@@ -17,6 +17,18 @@ class UserAccountRowProvisioner extends \LORIS\Data\Provisioners\DBRowProvisione
...
+        $user    = $factory->user();
+        if (!$user->hasPermission('user_accounts_multisite')) {
+            $siteLimitQuery = "LEFT JOIN user_psc_rel upsr2
+                ON (upsr2.CenterID=upsr.CenterID)
+                WHERE upsr2.UserID=:uid";
+            $params         = ['uid' => $user->getId()];
+        } else {
+            $siteLimitQuery = '';
+            $params         = [];
+        }
```
Comment (driusan) [inline modules/user_accounts/php/useraccountrowprovisioner.class.inc:31]: I don't understand why this is necessary.

user_accounts.class.inc already has:
```
    /**
     * Tells the base class that this page's provisioner can support the
     * HasAnyPermissionOrUserSiteMatch filter.
     *
     * @return ?array of permissions or null
     */
    public function allSitePermissionNames() : ?array
    {
        return ['user_accounts_multisite'];
    }
```

so shouldn't the HasAnyPermissionOrUserSiteMatch filter take care of it?

**Comment by CamilleBeau** [comment]
@jeffersoncasimir @driusan I removed all of the user_accounts code, changed the name, and the description. The PR now only includes the 2 functions that can be used elsewhere in the code.

**Pull request merged**